### ISSUE TITLE
Loss Suddenly Drop to 0 during some training experiments on Pixel200K

### ISSUE THREAD

**Issue by JingyuanHe1222** [bytedance/HLLM#15]
I was trying to reproduce the result on Pixel200K with TinyLlama 1.1B. 

One of my run matches the paper reported result, yet another run got a very weird loss and accuracy during training: 
The losses and the accuracy of the two runs were pretty much similar during the first 8800 steps of the first epoch, yet on step 8840, run 2 has a sudden drop in loss, and the training loss become 0 for the rest of the steps. 
The evaluation NDCG of this second run on the first epoch is very low (ndcg@10 ~0.0001) while in the first run which successfully reproduce the result, ndcg@10 was ~0.015. 

Here's the loss logs below. Do you have any intuition where goes wrong? I am thinking something mess up with the training data, yet I am using the same step up without modifying the data loading pipeline of HLLM and it got me why the second run messed up. 

```
Train [  0/  5]:  46%|████▋     | 8820/18971 [4:32:24<5:13:23,  1.85s/it, lr: 0.0000928 loss: 4.5625 data: 0.002 fwd: 0.612 bwd: 1.206 nce_samples: 108.013 nce_top1_acc: 0.237 nce_top5_acc: 0.312 nce_top10_acc: 0.400 nce_top50_acc: 0.675 nce_top100_acc: 0.788]03 Nov 01:19    INFO  
--------------------------------------------------

Train [  0/  5]:  47%|████▋     | 8840/18971 [4:33:01<5:13:02,  1.85s/it, lr: 0.0000930 loss: 0.2041 data: 0.002 fwd: 0.644 bwd: 1.229 nce_samples: 2.013 nce_top1_acc: 0.923 nce_top5_acc: 0.987 nce_top10_acc: 1.000 nce_top50_acc: 1.000 nce_top100_acc: 1.000]  03 Nov 01:20    INFO  
--------------------------------------------------

Train [  0/  5]:  47%|████▋     | 8860/18971 [4:33:38<5:11:29,  1.85s/it, lr: 0.0000932 loss: 0.0000 data: 0.002 fwd: 0.597 bwd: 1.193 nce_samples: 1.000 nce_top1_acc: 1.000 nce_top5_acc: 1.000 nce_top10_acc: 1.000 nce_top50_acc: 1.000 nce_top100_acc: 1.000]03 Nov 01:20    INFO  
--------------------------------------------------

Train [  0/  5]:  47%|████▋     | 8880/18971 [4:34:15<5:10:48,  1.85s/it, lr: 0.0000934 loss: 0.1357 data: 0.002 fwd: 0.636 bwd: 1.212 nce_samples: 1.609 nce_top1_acc: 0.971 nce_top5_acc: 1.000 nce_top10_acc: 1.000 nce_top50_acc: 1.000 nce_top100_acc: 1.000]03 Nov 01:21    INFO  
--------------------------------------------------

Train [  0/  5]:  47%|████▋     | 8900/18971 [4:34:52<5:10:23,  1.85s/it, lr: 0.0000936 loss: 0.0000 data: 0.002 fwd: 0.630 bwd: 1.212 nce_samples: 1.000 nce_top1_acc: 1.000 nce_top5_acc: 1.000 nce_top10_acc: 1.000 nce_top50_acc: 1.000 nce_top100_acc: 1.000]03 Nov 01:21    INFO  
--------------------------------------------------

Train [  0/  5]:  47%|████▋     | 8920/18971 [4:35:29<5:09:25,  1.85s/it, lr: 0.0000938 loss: 0.0000 data: 0.002 fwd: 0.598 bwd: 1.191 nce_samples: 1.000 nce_top1_acc: 1.000 nce_top5_acc: 1.000 nce_top10_acc: 1.000 nce_top50_acc: 1.000 nce_top100_acc: 1.000]03 Nov 01:22    INFO  
--------------------------------------------------

Train [  0/  5]:  47%|████▋     | 8940/18971 [4:36:06<5:08:41,  1.85s/it, lr: 0.0000940 loss: 0.0000 data: 0.002 fwd: 0.628 bwd: 1.205 nce_samples: 1.000 nce_top1_acc: 1.000 nce_top5_acc: 1.000 nce_top10_acc: 1.000 nce_top50_acc: 1.000 nce_top100_acc: 1.000]03 Nov 01:23    INFO  
```

Here's the config I used across two experiments: 

```
user_pretrain_dir="TinyLlama-1.1B-intermediate-step-1431k-3T"
item_pretrain_dir="TinyLlama-1.1B-intermediate-step-1431k-3T"
python3 main.py \
    --config_file /overall/LLM_deepspeed.yaml HLLM/HLLM.yaml \
    --loss nce \
    --epochs 5 \
    --dataset Pixel200K \
    --train_batch_size 8 \
    --MAX_TEXT_LENGTH 256 \
    --MAX_ITEM_LIST_LENGTH 10 \
    --checkpoint_dir $checkpoint_dir \
    --optim_args.learning_rate 1e-4 \
    --item_pretrain_dir $item_pretrain_dir \
    --user_pretrain_dir $user_pretrain_dir \
    --text_path $info_path \
    --data_path $inter_path \
    --text_keys '[\"title\",\"tag\",\"description\"]'
```

Thanks in advance!

**Comment by threestone965** [comment]
No matter what we input, the item model and the user model output the same embedding. At this time, the calculated loss is 0. I guess the model takes a shortcut. self.nce_thres should be the threshold used to filter positive samples from negative samples. You can try to set the variable to a higher precision, such as 0.999999

**Comment by ssyzeChen** [comment]
Hi, we also met this situation before especially with larger model. We also believe that some data in the datasets have interfered with the model's training. More negatives in the user LLM will help address this issue.

**Comment by JingyuanHe1222** [comment]
Hi, following up with your response, wonder if you observe this phenomenon in Pixel only? Or also in Amazon-Books?

**Comment by ssyzeChen** [comment]
> Hi, following up with your response, wonder if you observe this phenomenon in Pixel only? Or also in Amazon-Books?

Actually happened in all datasets before 😭
